src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -271,26 +283,115 @@ private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(B
             conversionDiagnostics?.Free();
 
             var intType = GetSpecialType(SpecialType.System_Int32, diagnostics, unconvertedInterpolatedString.Syntax);
-            var arguments = ImmutableArray.Create<BoundExpression>(
-                new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(baseStringLength), intType) { WasCompilerGenerated = true },
-                new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(numFormatHoles), intType) { WasCompilerGenerated = true });
+            int constructorArgumentLength = 2 + additionalConstructorArguments.Length;
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(constructorArgumentLength);
+            populateArguments(unconvertedInterpolatedString.Syntax, additionalConstructorArguments, baseStringLength, numFormatHoles, intType, argumentsBuilder);
+
+            var refKindsBuilder = ArrayBuilder<RefKind>.GetInstance(constructorArgumentLength);
+            refKindsBuilder.Add(RefKind.None);
+            refKindsBuilder.Add(RefKind.None);
+            refKindsBuilder.AddRange(additionalConstructorRefKinds);
+
+            var nonOutConstructorDiagnostics = BindingDiagnosticBag.GetInstance(withDiagnostics: true, withDependencies: diagnostics.AccumulatesDependencies);
+            BoundExpression constructorCall;
+            BoundExpression nonOutConstructorCall = MakeConstructorInvocation(interpolatedStringHandlerType, argumentsBuilder, refKindsBuilder, unconvertedInterpolatedString.Syntax, nonOutConstructorDiagnostics);
+            if (nonOutConstructorCall is not BoundObjectCreationExpression { ResultKind: LookupResultKind.Viable })
```

Comment:
nit: consider inverting `if` by removing `not` from pattern

Reply:
I think it reads better as is.